python_hw_1.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/python 
+
+
+def transcribe(seq):
+    """Prints the transcribed sequence"""
+    transcription_seq = ""
+    transcription_dic = {"A" : "U", "T" : "A", "G" : "C", "C" : "G", 
+                "a" : "u", "t" : "a", "g" : "c", "c" : "g"}
+    for letter in seq:
+        if letter not in transcription_dic.keys():
+            print("Invalid alphabet")
+            break
+        else:
+            transcription_seq += transcription_dic[letter]
+    print(transcription_seq)
+    return transcription_seq 
+
+def reverse(seq):
+    """Prints the reverse sequence"""
+    rev_seq = ""
+    for letter in seq:
+        rev_seq = seq[::-1]
+    print(rev_seq)
+    return rev_seq 
+
+def complement(seq):
+    """Prints the complement sequence"""
+    compl_seq = ""
+    compl_dic = {"A" : "T", "T" : "A", "G" : "C", "C" : "G", 
+                "a" : "t", "t" : "a", "g" : "c", "c" : "g"}
+    for letter in seq:
+        compl_seq += compl_dic[letter]
+    print(compl_seq)
+    return compl_seq
+
+def reverse_complement(seq):
+    """Prints the reverse complement sequence"""
+    compl = ""
+    rev_compl = ""
+    compl_dic = {"A" : "T", "T" : "A", "G" : "C", "C" : "G", 
+                 "a" : "t", "t" : "a", "g" : "c", "c" : "g"}
+    for letter in seq:
+        compl += compl_dic[letter]
+        rev_compl = compl[::-1]
+    print(rev_compl)
+    return rev_compl
+
+def check_letters(seq): 
+    nucl = ["U", "T"]
+    up_seq = seq.upper()
+    for letter in seq:
+        result = all(letter in up_seq for letter in nucl)
+        if result:
+            return False
+    return True
```

Comment:
Мы не проходили, но такие штуки очень удобно делать при помощи множеств
```suggestion
    nucl = {"U", "T"}
    return nucl <= set(seq.upper())

```
<= - оператор проверки на подмножество